src/main/java/com/exasol/adapter/dynamodb/documentpath/DocumentPathIteratorFactory.java

```diff
@@ -32,4 +34,13 @@ public DocumentPathIteratorFactory(final DocumentPathExpression path, final Docu
             return new LoopDocumentPathIterator<>(this.path, this.document);
         }
     }
+
+    /**
+     * Gives a stream of iteration states.
+     * 
+     * @return stream of {@link PathIterationStateProvider}
+     */
+    public Stream<PathIterationStateProvider> stream() {
+        return StreamSupport.stream(spliterator(), false);
```

Comment:
Do you have to override the `splititerator` or the default from iterator okay here?

Reply:
The default onw is:
````
return Spliterators.spliteratorUnknownSize(this.iterator(), 0);
```
That fits here